draft/XXXX-missing-values-in-typed-cols/README.md

```diff
@@ -0,0 +1,44 @@
+
+There will likely be some client- and server-side considerations for handling missing data. This will include correctly parsing data and when reading it into the ArangoDB types (or we may choose to leave them as strings in the db). Types that we define, will be referenced as semantic types, and native ArangoDB types will be referenced as Arango types
+
+## Motivation
+
+We now have a motivation to supply data schema information to visualization applications, UpSet. Upset requires set membership information and other data types to generate its visualization. We can also utilize this type information in the other visualization applications that support multinet (MultiLink and MultiMatrix).
+
+Since we now need to pass semantic typing into these applications, knowing what the types are, even if there is missing data, is essential. As such, we'll need to modify our typing logic to handle missing data. This stretches from the frontend, where we provide type suggestions, to the backend, where we store the type information and disseminate it to the vis applications.
+
+## Proposals
+
+We should provide type information for columns that are of one type, but that have missing values. There are at least 3 places that we should support this paradigm:
+
+1. `multinet-client` on data preview/type suggestion.
+2. `multinet-server` on data upload and data fetching.
+3. `multinetjs` on data upload and data fetching.
+
+For the type suggester, there is a code chunk in the [Reference Implementation](#reference-implementation) section that suggests a way that we can check for the type of a column, even when data is missing.
+
+For the other repos, `multinet-server` and `multinetjs`, the considerations are slightly different. We'll need to think about how to store and serve data with missing values, and how to disseminate the type information given that there are missing values. Additionally, an open question is, should we tell client applications to expect some missing data in typed columns? This could help when rendering visualizations.
+
+## Backwards Compatibility
+
+There will be no added backwards incompatibilities from this change. There may be some knock-on consequences for apps that can utilize types in the case that there was data uploaded with the old type definitions, but the impact will be minimal.
+
+## Reference Implementation
+
+The client already has logic to calculate the type of the columns but does not allow for missing values as it stands. This implementation should just be an extension of the existing logic.
+
+Jake gave a suggestion for how we might calculate if a column is actually the correct type with some missing values:
+
+```
+const number = entry.number + entry.empty === entry.total;
+```
```

Comment:
Beyond this, we can also display something when `entry.empty > 0` to indicate that the column has missing values (if the uploader intended for this not to be the case, it would be a useful prompt to look at their raw data again).

Reply:
I like this idea. I'll make sure I note it down in the RFC

Reply:
Updated in 4493627

Reply:
We should aim for decisiveness in these RFCs--we want to use them as a guide to implementation after they're settled. So if these feature requires further debate, let's do it and decide one way or the other as much as possible.

In this case, I agree that the client could check for missing data itself and use that fact to present an indicator in the UI. The API may not need to be involved at all. If you feel the same way, go ahead and resolve the language to reflect that.

(Sometimes it may really be impossible to decide until implementation time how something should be, and in those cases of course it's ok to make a note of that in the RFC. But this doesn't seem to be one of those times.)

Reply:
0d938f8